src/utils.ts

```diff
@@ -0,0 +1,34 @@
+const stringToBoolean = (value: string) => {
+  const stringBooleans = ["true", "false"];
+  if (value && stringBooleans.includes(value)) {
+    return value === "true" ? true : false;
+  }
+  return value;
+};
+
+const stringToNumber = (value: string) => {
+  const number = Number(value);
+  if (isNaN(number)) return value;
+  return number;
+};
+
+export const parseUserData = (data: any) => {
```

Comment:
Could you define the output type to avoid issues?

```suggestion
export const parseUserData: (data: any) => UserData = (data: any) => {
```